### PR TITLE
fix: no need to filter filterdKeys for custom filterDropdown.

### LINKED ISSUE #28603: Filters doesn't pass into Table onChange handler
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-sb8w9?file=/index.js)

### Steps to reproduce
1. Click to `name` column filter
2. Type something
3. Click Search
4. Go to the console

### What is expected?
`name` filters includes search string

### What is actually happening?
`name` filters are empty

| Environment | Info |
|---|---|
| antd | 4.9.4 |
| React | 17.0.1 |
| System | Windows 10 |
| Browser | Chrome  87.0.4280.88 |

---
This error appeared after https://github.com/ant-design/ant-design/pull/28220 in function `generateFilterInfo`

### LINKED ISSUE #28640: Table onChange filters is empty when using custom filters
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/customized-filter-panel-antd494-forked-s63ji?file=/index.js)

### Steps to reproduce
1. Try to filter by Name
2. Check console name is empty https://prnt.sc/we2gb3

### What is expected?
Name should have filtered value https://prnt.sc/we2gb3

### What is actually happening?
filtered value is empty in name https://prnt.sc/we2gb3

| Environment | Info |
|---|---|
| antd | 4.9.4 |
| React | 16.14.0 |
| System | OSX |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by Meowu**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
fix #28603
fix #28640
close #28541

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     fix that onChange listener always receivces empty list as `filters` parameter if use a custom filterDropdown     |
| 🇨🇳 Chinese |    修复了如果用户自定义 filterDropdown， onChange 事件的 `filters` 参数总是接收空数组的问题       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Review by Edifear — APPROVED** [review]

File: components/table/__tests__/Table.filter.test.js [hunk 1]
```diff
@@ -453,7 +453,7 @@ describe('Table.filter', () => {
     const wrapper = mount(createTable({ onChange: handleChange }));
 
     wrapper.find('.ant-dropdown-trigger').first().simulate('click');
-    wrapper.find('.ant-table-filter-dropdown-btns .ant-btn-link').simulate('click');
+    wrapper.find('.ant-table-filter-dropdown-btns .ant-btn-primary').simulate('click');
```
Comment (afc163) [inline components/table/__tests__/Table.filter.test.js:456]: 为啥改这个。

**Comment by afc163** [comment]
rebase 一下 master。

**Comment by afc163** [comment]
统一解决一下 https://github.com/ant-design/ant-design/issues/28640 和 https://github.com/ant-design/ant-design/pull/28541 吧。

File: components/table/__tests__/Table.filter.test.js (hunk 1 above)
Comment (Meowu) [reply components/table/__tests__/Table.filter.test.js:456]: > 为啥改这个。

这里 link 应该是 reset 按钮，primary 是 OK 按钮，如果不选任何 filter 的情况下，reset 按钮应该是 disabled 的吧，那点击它应该是不起作用的。

**Comment by Meowu** [comment]
> 统一解决一下 #28640 和 #28541 吧。

已解决。

File: components/table/hooks/useFilter/index.tsx [hunk 2]
```diff
@@ -34,7 +34,9 @@ function collectFilterStates<RecordType>(
     } else if (column.filters || 'filterDropdown' in column || 'onFilter' in column) {
       if ('filteredValue' in column) {
         // Controlled
-        const filteredValues = (column.filteredValue || []).map(String);
+        const filteredValues = Array.isArray(column.filteredValue)
+          ? column.filteredValue.map(String)
```
Comment (seta-namle) [inline components/table/hooks/useFilter/index.tsx:38]: @Meowu @afc163  In my case, the custom autocomplete filter drop-down would expect an array of objects ([ {key: 1, label: 'Item 1'}, {key: 2, label: 'Item 2'}]). It resembles  a controlled value of autocomplete component which has labelInValue prop. This code converted value to **["[object Object]", "[object Object]"]** :cry:

Comment (Meowu) [reply components/table/hooks/useFilter/index.tsx:38]: orz..
I didn't think about this edge case.It's complicated.Would you mind providing a sample code.

Comment (seta-namle) [reply components/table/hooks/useFilter/index.tsx:38]: @Meowu I have reproduced a simple example.
https://codesandbox.io/s/filter-and-sorter-antd4100-forked-wfkx5

Comment (seta-namle) [reply components/table/hooks/useFilter/index.tsx:38]: In my other case, the onChange event of the table isn't triggered when the type of filtered value is object. [ click me](https://prnt.sc/wgiscm)

Comment (Meowu) [reply components/table/hooks/useFilter/index.tsx:38]: OK.I will do more test and fix it.

Comment (Meowu) [reply components/table/hooks/useFilter/index.tsx:38]: will fix in https://github.com/ant-design/ant-design/pull/28850 .

**Pull request merged**